### PR TITLE
enc, cleaning of secret data 

### PR DESCRIPTION
**Pull request by EasySec**
Cleaning key data as per @dot-asm comment on PR #4488

### CODE REVIEW

File: apps/enc.c [hunk 1]
```diff
@@ -476,9 +476,14 @@ int enc_main(int argc, char **argv)
...
             goto end;
         }
-        if ((hkey != NULL) && !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
-            BIO_printf(bio_err, "invalid hex key value\n");
-            goto end;
+        if (hkey != NULL) {
+            if( !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
+                BIO_printf(bio_err, "invalid hex key value\n");
+                goto end;
+            }
```
Comment (paulidale) [inline apps/enc.c]: Small style NIT.  The `else {` should be on the same line as the closing for the _then_ body.  I.e. `} else {`

Comment (EasySec) [reply apps/enc.c]: Yes, thanks ! Done.

File: apps/enc.c [hunk 2]
```diff
@@ -476,9 +476,13 @@ int enc_main(int argc, char **argv)
             BIO_printf(bio_err, "iv undefined\n");
             goto end;
         }
-        if ((hkey != NULL) && !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
-            BIO_printf(bio_err, "invalid hex key value\n");
-            goto end;
+        if (hkey != NULL) {
+            if( !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
```
Comment (dot-asm) [inline apps/enc.c]: Formatting nit, it's `if (!`, not `if( !`.

**Comment by dot-asm** [comment]
What's up with red cross from travis? Well, formatting nit would trigger another test, so we'll see it it's intermittent... Should/has to be...

File: apps/enc.c (hunk 2 above)
Comment (EasySec) [reply apps/enc.c]: Sorry . Done.

**Comment by richsalz** [comment]
Squashed and merged.  Thank you!

**Pull request closed without merging**